app/src/main/java/org/dhis2/usescases/teiDashboard/DashboardRepositoryImpl.java

```diff
@@ -374,12 +373,34 @@ public void setDefaultCatOptCombToEvent( String eventUid )
     {
         if ( programUid != null )
             return briteDatabase
-                .createQuery( ATTRIBUTE_VALUES_TABLE, ATTRIBUTE_VALUES_QUERY, programUid, teiUid == null ? "" : teiUid )
-                .mapToList( cursor -> ValueUtils.transform( briteDatabase, cursor ) );
+                    .createQuery( ATTRIBUTE_VALUES_TABLE, ATTRIBUTE_VALUES_QUERY, programUid, teiUid == null ? "" : teiUid )
+                    .mapToList( cursor -> ValueUtils.transform( briteDatabase, cursor ) )
+                    .map(values -> {
+                        List<TrackedEntityAttributeValue> toRemove = new ArrayList<>();
+                        for (TrackedEntityAttributeValue value: values) {
+                            ValueType type = d2.trackedEntityModule().trackedEntityAttributes().uid(value.trackedEntityAttribute()).blockingGet().valueType();
+                            if (type == ValueType.IMAGE) {
+                                toRemove.add(value);
+                            }
+                        }
+                        values.removeAll(toRemove);
+                        return values;
```

Comment:
Do we need to perform this in the IO?

Reply:
I believe the IO is the perfect place to do database operations within the repository.

Reply:
yes I agree !!! my question was the I did not see scheduler.io in any place, is this being done inside the sdk?